diff --git a/data/com.feralinteractive.GameMode.policy.in b/data/com.feralinteractive.GameMode.policy.in
index 9933e0b..611b4dc 100644
--- a/data/com.feralinteractive.GameMode.policy.in
+++ b/data/com.feralinteractive.GameMode.policy.in
@@ -20,7 +20,7 @@
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
     </defaults>
-    <annotate key="org.freedesktop.policykit.exec.path">@LIBEXECDIR@/cpugovctl</annotate>
+    <annotate key="org.freedesktop.policykit.exec.path">/run/current-system/sw/bin/gamemode-daemon/cpugovctl</annotate>
   </action>
 
   <action id="com.feralinteractive.GameMode.gpu-helper">
@@ -31,7 +31,7 @@
       <allow_inactive>no</allow_inactive>
       <allow_active>yes</allow_active>
     </defaults>
-    <annotate key="org.freedesktop.policykit.exec.path">@LIBEXECDIR@/gpuclockctl</annotate>
+    <annotate key="org.freedesktop.policykit.exec.path">/run/current-system/sw/bin/gamemode-daemon/gpuclockctl</annotate>
     <annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
   </action>
 
