diff --git a/lisp/emacs-lisp/comp.el b/lisp/emacs-lisp/comp.el
--- a/lisp/emacs-lisp/comp.el
+++ b/lisp/emacs-lisp/comp.el
@@ -188,7 +188,8 @@
 
 (defcustom native-comp-driver-options
   (cond ((eq system-type 'darwin) '("-Wl,-w"))
-        ((eq system-type 'cygwin) '("-Wl,-dynamicbase")))
+        ((eq system-type 'cygwin) '("-Wl,-dynamicbase"))
+        (t '(@backendPath@)))
   "Options passed verbatim to the native compiler's back-end driver.
 Note that not all options are meaningful; typically only the options
 affecting the assembler and linker are likely to be useful.
